cmake/RocksDBConfig.cmake.in

```diff
@@ -1,3 +1,56 @@
 @PACKAGE_INIT@
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/modules")
+
+include(CMakeFindDependencyMacro)
+
+set(GFLAGS_USE_TARGET_NAMESPACE @GFLAGS_USE_TARGET_NAMESPACE@)
+
+if(@WITH_JEMALLOC@)
+  find_dependency(JeMalloc)
+endif()
+
+if(@WITH_GFLAGS@)
+  find_dependency(gflags CONFIG)
+  if(NOT gflags_FOUND)
+    message("GFlags config not found. Fallback to legacy method.")
+    find_dependency(gflags)
```

Comment:
why not use `find_package()` 's builtin fallback machinery?

quote from cmake's document:

> find_package(<PackageName> [version] [EXACT] [QUIET] [MODULE]
>              [REQUIRED] [[COMPONENTS] [components...]]
>              [OPTIONAL_COMPONENTS components...]
>              [NO_POLICY_SCOPE])
> ...
> If the MODULE option is not specfied in the above signature, CMake first searches for the package using Module mode. Then, if the package is not found, it searches again using Config mode. A user may set the variable CMAKE_FIND_PACKAGE_PREFER_CONFIG to TRUE to direct CMake first search using Config mode before falling back to Module mode.

Reply:
Because find_package() does MODULE>CONFIG (i.e. prefer user overwriting behavior) while we're looking for CONFIG>MODULE (i.e. provide FindXXX when official XXXConfig is missing)

Reply:
i must be missing something.. what's wrong with MODULE > CONFIG ? if our `Find*.cmake` modules align with the config files offered by upstream?

also, as your commit message is practically empty. would be better if you could amend it with more details regarding "why" we need to have this change.

Reply:
It's because aligning with upstream is usually unrealistic given upstream can change and there're so many versions of "upstreams".
Also you may noticed the upstream config is very complex, for handling all kinds of build options and with backward compatibility. I don't think it's trivial to mimic upstream behavior.

Reply:
Commit msg updated.